mgnify_pipelines_toolkit/analysis/assembly/mgnify2mimc_inputs.py

```diff
@@ -0,0 +1,726 @@
+                        ],
+                    )
+
+                    pfam_clan_df.rename(
+                        columns={"Protein_family": "PfamID"}, inplace=True
+                    )
+
+        except requests.exceptions.RequestException as e:
+            print(
+                f"Pfam clan download failed due to error: {e}. Please try again later"
+            )
+
+    else:
+        print(
+            "This version of pfam is not compatible with the mgnify2mimic_inputs pipeline."
+        )
+        print(
+            "Compatible versions: 32.0, 33.0, 33.1, 34.0, 35.0, 36.0, 37.0, 37.1, 37.2"
+        )
+        sys.exit(1)
+
+    return pfam_clan_df
+
+
+def generate_pfam_metagenome_vector(pfam_clan_df, samples_pfam_dict, pfam_vector_file):
+
+    # get list of pfamIDs
+    pfam_clan_ids = list(pfam_clan_df["PfamID"])
+
+    # assign pfam dict
+    pfam_vector_df = pfam_clan_df[["PfamID"]]
+
+    title = ["PfamID"]
+    title.extend(list(samples_pfam_dict.keys()))
+    title = "\t".join(title) + "\n"
+
+    print("Generating pfam vector for metagenome")
+    try:
+        with open(pfam_vector_file, "w") as rfile:
+            rfile.write(title)
+            for pfam in pfam_clan_ids:
+                rline = [
+                    "1" if pfam in samples_pfam_dict[mgya] else "0"
+                    for mgya in samples_pfam_dict
+                ]
+                rline.insert(0, pfam)
+
+                # print (rline)
+
+                rline = "\t".join(rline) + "\n"
+
+                rfile.write(rline)
+
+    except FileNotFoundError:
+        print(
+            f"metagenome file path: {pfam_vector_file} is not correct. Please check check filepath and retry command."
+        )
+
+    return pfam_vector_df
+
+
+# for a given biome, download the functional profiles from the associated catalogue and parse the pfams into a vector
+def download_genome_functions(biome, download_directory):
+
+    biomes_dict = {
+        "chicken-gut": "v1.0.1",
+        # "cow-rumen": "v1.0.1",
+        "honeybee-gut": "v1.0.1",
+        # "human-gut":"v2.0.2",
+        # "human-oral":"v1.0.1",
+        "human-vaginal": "v1.0",
+        "marine": "v2.0",
+        "mouse-gut": "v1.0",
+        "non-model-fish-gut": "v2.0",
+        # "pig-gut":"v1.0",
+        "sheep-rumen": "v1.0",
+        "zebrafish-fecal": "v1.0",
+    }
+
+    biomes = ", ".join(biomes_dict.keys())
+
+    # if biome is not in biomes_dict (i.e. pangenomes functions doesn't exist at the moment for the specified catalogue)
+    if biome not in biomes_dict:
+        print(
+            f"*** specified biome: no pfam files exist for the {biome} catalogue. Please use one from the following: {biomes}. ***"
+        )
+    else:
+        catalogue_version = biomes_dict[biome]
+
+    ftp_url = (
+        f"https://ftp.ebi.ac.uk/pub/databases/metagenomics/mgnify_genomes/{biome}/"
+        f"{catalogue_version}/pangenome_functions/functional_profiles.tar.gz"
```

Comment:
Update the url to the following:
`https://ftp.ebi.ac.uk/pub/databases/metagenomics/pipelines/references/mgnify_genomes/`